### PR TITLE
Update to MetalLB v0.10.3

### PR DESCRIPTION
**Pull request by majst01**
- Update metallb from 0.8.3 -> v0.10.3, k8s v1.21 compatibility

TODO:
- [x] test in real infra (additional integration tests required ? )
- [x] update to latest release
- [x] keep running in test for couple of days/weeks before merging

Interesting hint from the metallb issues regarding metallb and calico interoperability: https://github.com/metallb/metallb/issues/986

## Current findings:

With the update the metallb-speaker will be restarted, but because the bgp daemon on the worker node is configured with:
```
 neighbor LOCAL peer-group
 neighbor LOCAL remote-as internal
 neighbor LOCAL timers 1 3
```
Existing announcements are withdrawn from the bgpd once the pod is not running anymore. 
This is the case as long the new image is pulled and started. This will take ~20-30sec. During this time the external services will not be available, existing connections are not dropped, the connection simply stalls. 
But only if:
- one worker
- externaltrafficpolicy=local and service is schedule on one node
- service only scheduled on one node

Could be mitigated by pre pulling the metallb container image, then downtime will be reduced to 5-10sec.

Closes https://github.com/metal-stack/gardener-extension-provider-metal/issues/161.

## Release Notes

```ACTIONS_REQUIRED
In our next releases we are planning to update MetalLB to v0.10.3. During the rolling update of the `speaker` DaemonSet, a cluster worker node withdraws its BGP route announcements for services of type `LoadBalancer` for as long as it takes to pull the new image and start the MetalLB `speaker` pod again (normally takes less than 30 seconds). 
This process can lead to service unavailability if one of the following conditions are met:

- Your cluster only has a single worker node (such that another worker cannot overtake ingress traffic, which is distributed in your cluster through `kube-proxy`)
- Your service is only scheduled on a single worker node and you set the `externaltrafficpolicy` in your service to `local` (https://metallb.universe.tf/usage/#local-traffic-policy-1)

Existing connections are not dropped, the connection just stalls.

🔧 Please check your clusters for these conditions and take action if you want to mitigate service unavailability.
```

### CODE REVIEW

**Comment by Gerrit91** [comment]
Closes #161.

File: charts/internal/shoot-control-plane/templates/metallb.yaml [hunk 1]
```diff
@@ -226,20 +388,19 @@ spec:
...
           capabilities:
             add:
-            - NET_ADMIN
             - NET_RAW
-            - SYS_ADMIN
             drop:
             - ALL
           readOnlyRootFilesystem: true
-      hostNetwork: false
+      hostNetwork: true
```
Comment (Gerrit91) [inline charts/internal/shoot-control-plane/templates/metallb.yaml:395]: Why did this change?

Comment (majst01) [reply charts/internal/shoot-control-plane/templates/metallb.yaml:395]: Just copied over mostly the manifest from upstream, but good catch

**Comment by majst01** [comment]
pipeline fails for unknown reason ??

File: charts/internal/shoot-control-plane/templates/metallb.yaml (hunk 1 above)
Comment (majst01) [reply charts/internal/shoot-control-plane/templates/metallb.yaml]: reset to false

**Comment by Gerrit91** [comment]
There was a 0.10.3 release yesterday, we should use this version instead.

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**